source/Calamari/Plugins/PluginLoader.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Calamari.Commands.Support;
+
+namespace Calamari.Plugins
+{
+    public static class PluginLoader
+    {
+        public static (ICalamariPlugin[] plugins, string[] remainingOptions) Load(string[] commandLineArgs)
+        {
+            var (paths, remainingOptions) = GetPluginPaths(commandLineArgs);
+            var plugins = LoadAssemblies(paths)
+                .SelectMany(a => a.GetTypes())
+                .Where(t => typeof(ICalamariPlugin).IsAssignableFrom(t))
+                .Select(t => (ICalamariPlugin) Activator.CreateInstance(t))
+                .ToArray();
+            
+            return (plugins, remainingOptions);
+        }
+
+        private static List<Assembly> LoadAssemblies(string[] paths)
+        {
+            var load = new List<Assembly>();
+            foreach (var path in paths)
+            {
+                try
+                {
+                    load.Add(Assembly.LoadFrom(path));
```

Comment:
So long as we develop and update all our plugins in lockstep this will work, otherwise we need to be careful about needing to load different versions of dependencies which leads to needing to use `AssemblyLoader` etc

Reply:
given that these plugins will be _ours_ unlike our thoughts a year or two ago, i think its fair to assume we can control the dependencies properly which will allow this to work 👍